### DISCUSSION TITLE
flux healthchecks fail for daemonset with rollingupdate strategy

### DISCUSSION THREAD

**Discussion by KNSailaja** [fluxcd/flux2#4093] [General]
We have a daemonset running on our nodes and we use kustomization healthchecks on daemonset. During rollingupdate there is atleast one pod down and thats is reporting kustomization as unhealthy . Is there any way in flux to hold on until rolling update is finished before marking kustomization as unhealthy.

**Comment by somtochiama** [comment]
You can increase the `spec.timeout` on the Kustomization so that it waits a longer time for the Daemonset to be ready.

**Reply by KNSailaja** [reply]
Thanks. wl test it with increased timeout

**Reply by KNSailaja** [reply]
Also, Do you know the criteria for a daemonset to be considered healthy?

**Reply by somtochiama** [reply]
kstatus looks at the status of the daemosets and compare the `status.desiredNumberScheduled` field with `status.currentNumberScheduled`, `status.numberAvailable` and `.status.numberReady`
